skxray/img_proc/threshops.py

```diff
@@ -0,0 +1,324 @@
+# Module for the BNL image processing project
+# Developed at the NSLS-II, Brookhaven National Laboratory
+# Developed by Gabriel Iltis, Oct. 2013
+"""
+This module contains tools for thresholding data sets.
+"""
+
+"""
+REVISION LOG: (FORMAT: "PROGRAMMER INITIALS: DATE -- RECORD")
+GCI: 2/11/2014 -- Modifying documentation of the package functions for 
+    inclusion in the bulk module pull to GITHUB
+GCI: 2/19/2014 -- Renaming module from C4_threshops to threshops.py
+    Updating documentation to docstring convention.
+    Changed module back from class structure to a simple module containing
+    thresholding function definitions.
+GCI: 9/23/14 -- Modified all functions to auto-wrap into the image
+    processing library in vistrails.
+"""
+
+import numpy as np
+import skimage.filter as sk
+
+
+def thresh_globalGT(src_data, thresh_value, md_dict=None):
+    """
+    This function is a basic thresholding operation allowing for isolation, or 
+    selection, of all voxels having an intensity value greater than the 
+    specificed threshold value. The result is a binary array of equal size 
+    containing voxel values of either 0 or 1.
+
+    Parameters
+    ----------
+    src_data : ndarray
+        Specify the volume to be thresholded
+    
+    thresh_value : float
+        Specify the threshold value, above which contains the desired voxel 
+        values to be isolated. Input value is of type float, however,
+        the dtype of this parameter will be automatically converted to the
+        dtype of the source data if the source data is not of type float.
+
+    md_dict : dict, optional
+        Metadata dictionary for the data set(s) being segmented or analyzed,
+        or for the analysis being conducted
+
+    Returns
+    -------
+    output : ndarray
+        The function returns a binary array where all voxels with values equal 
+        to 1 correspond to voxels within the identified threshold range.
+    """
+
+    if thresh_value > np.amax(src_data):
+        raise ValueError("Selected threshold value is greater than the "

```

Comment:
```python
thresh_val = XXX
for img in list_of_tenk_images:
    thresh_globalGT(img, thresh_val)
```

Having this blow up half way through because you have one frame where all of the value drop above or below your threshold does not seem like the right thing to do.  There are use-cases where you may want to check that you are not saturating, but that is easy enough to be checked on the user side.

I really don't see the advantage of this over `np.greater` which is already wrapped in now.

Another part of the curation process is making sure we don't include too many very similar functions which will only serve to over-whelm and confuse the users.

If  this behavior stays it needs to have very prominent warnings in the docstring.